<?xml version="1.0"?>

<launch>
  <arg name="twist_control" default="false" />

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find audibot_control)/config/audibot_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node ns="/audibot" name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="audi_speed_controller audi_steering_controller joint_state_controller --shutdown-timeout 0.5"/>

  <node if="$(arg twist_control)" ns="/audibot" name="twist_controller" pkg="audibot_control" type="audi_twist_controller" output="screen" >
    <rosparam file="$(find audibot_control)/yaml/twist_controller_params.yaml" />
  </node>

</launch>
